Adnan/chrome bok choy fixes

adnanghaffar07 opened the pull request.

adnanghaffar07 commented:
@benpatterson @jzoldak @raeeschachar 

This PR have 6 tests with fixes on GC and FF.

1-  lms.test_library.LibraryContentTest.test_shows_all_if_max_set_to_greater_value
2-  lms.test_library.LibraryContentTest.test_shows_random_xblocks_from_configured_1_2
3- lms.test_library.LibraryContentTest.test_shows_random_xblocks_from_configured_2_3
4- lms.test_library.LibraryContentTest.test_shows_random_xblocks_from_configured_3_4
5- lms.test_lms_problems.ProblemPartialCredit.test_partial_credit
6- lms.test_library.StudioLibraryContainerCapaFilterTest.test_problem_type_selector	

For test_shows_all_if_max_set_to_greater_value: on Chrome it was not clearing the values and appending with current values.
Solution: I have made the two changes for this, select all and then delete the values. It works good on Chrome and FF.

For test_shows_random_xblocks_from_configured_1_2: On Chrome it doesn't consider change if we delete the value and then write same value e.g If we delete the 1 and then write the same value 1. Secondly, it was not clearing the values.
Solution: I did change the test data and it works good on Chrome and FF. Both issues have been fixed.

For test_shows_random_xblocks_from_configured_2_3: I made the changes in data because It doesn't make sense if we run the same test two times with same data. Secondly, it was not clearing the values. It is passing on chrome and firefox.

For test_shows_random_xblocks_from_configured_3_4: I made the changes in test data and in fixtures.  Secondly, it was not clearing the values. Now, is passing on chrome and firefox.

For test_partial_credit: It was failing because chrome run fast and status was not appearing for assert. 
Solution: I have added the two wait and 1 function for status icon. 

test_problem_type_selector is fixed with above fixes and It is working on Chrome now.

benpatterson commented on common/test/acceptance/pages/lms/problem.py:
```diff
@@ -19,6 +19,7 @@ def problem_name(self):
         """
         Return the current problem name.
         """
+        self.wait_for_element_presence('.problem-header', 'wait for problem header')
```
This kind of statement should be in the test, not the property. Wherever you are using this property, you should have the wait statement right before it. That, or the is_brower_on_page for that page needs to be smarter.

benpatterson commented on common/test/acceptance/tests/lms/test_library.py:
```diff
@@ -156,9 +157,11 @@ def populate_library_fixture(self, library_fixture):
             XBlockFixtureDesc("html", "Html2", data='html2'),
             XBlockFixtureDesc("html", "Html3", data='html3'),
+            XBlockFixtureDesc("html", "Html4", data='html4'),
+
```
Pls remove this extra line

adnanghaffar07 commented:
@benpatterson @jzoldak @raeeschachar 

I have addressed the Ben's feedback and corrected the code. 
Please review, I have added wait statements in test and removed the extra lines.

benpatterson commented:
:+1:

benpatterson commented:
Please squash your commits before you merge.

raeeschachar commented:
:+1:
